test/Microsoft.Extensions.Configuration.UserSecrets.Test/ConfigurationExtensionTest.cs

```diff
@@ -82,68 +80,55 @@ public void AddUserSecrets_FindsAssemblyAttributeFromType()
         }
 
         [Fact]
-        public void AddUserSecrets_ShowsAssemblyAttributeError_When_ProjectJson_Missing()
+        public void AddUserSecrets_With_SecretsId_Passed_Explicitly()
         {
-            var projectPath = _fixture.GetTempSecretProject();
-            File.Delete(Path.Combine(projectPath, "project.json"));
-
-            var builder = new ConfigurationBuilder().SetBasePath(projectPath);
+            var userSecretsId = Guid.NewGuid().ToString();
+            SetSecret(userSecretsId, "Facebook:AppSecret", "value1");
 
-#pragma warning disable CS0618
-            var ex = Assert.Throws<InvalidOperationException>(() => builder.AddUserSecrets());
-#pragma warning restore CS0618
-            Assert.Equal(Resources.FormatError_Missing_UserSecretsIdAttribute(Assembly.GetEntryAssembly().FullName), ex.Message);
-        }
+            var builder = new ConfigurationBuilder().AddUserSecrets(userSecretsId);
+            var configuration = builder.Build();
 
-        [Fact]
-        public void AddUserSecrets_Does_Not_Fail_On_Non_Existing_File_Explicitly_Passed()
-        {
-            var builder = new ConfigurationBuilder()
-                                .AddUserSecrets(userSecretsId: Guid.NewGuid().ToString());
+            Assert.Equal("value1", configuration["Facebook:AppSecret"]);
         }
 
         [Fact]
         public void AddUserSecrets_Does_Not_Fail_On_Non_Existing_File()
         {
             string userSecretsId;
-            var projectPath = _fixture.GetTempSecretProject(out userSecretsId);
-#pragma warning disable CS0618
-            var builder = new ConfigurationBuilder().SetBasePath(projectPath).AddUserSecrets();
-#pragma warning restore CS0618
-
-            var configuration = builder.Build();
-            Assert.Equal(null, configuration["Facebook:AppSecret"]);
-        }
-
-        [Fact]
-        public void AddUserSecrets_With_An_Existing_Secret_File()
-        {
-            string userSecretsId;
-            var projectPath = _fixture.GetTempSecretProject(out userSecretsId);
-
-            SetSecret(userSecretsId, "Facebook:AppSecret", "value1");
+            string secretFilePath;
+            for (;;)
```

Comment:
Probably unnecessary. Was thinking about SHA1 collisions when I wrote this :) I'll remove as its highly unlikely guids will collide.